include/base/dof_map.h

```diff
@@ -498,6 +502,8 @@ class DofMap : public ReferenceCountedObject<DofMap>,
    */
   const std::vector<dof_id_type> & get_send_list() const { return _send_list; }
 
+  const std::shared_ptr<GlobalToLocalMap> get_global_to_local_map() const { return _global_to_local_map; }
```

Comment:
This still lets us modify _global_to_local_map from other classes, doesn't it?  Since ```shared_ptr<T>``` seamlessly converts to ```shared_ptr<const T>```, shouldn't we return the latter here?

Reply:
We can not change this. In pets_vector.h, we have 

`std::shared_ptr<GlobalToLocalMap> _global_to_local_map;`

It need to be compatible with the old API. If send_list is used, we need to add indices into `_global_to_local_map`.

If `_global_to_local_map` points to a constant object, we can not do any modifications

Reply:
Sounds like it should not be a `const` function in that case.